pkg/crypto/crypto.go

```diff
@@ -29,10 +29,11 @@ func EncryptVolume(devicePath, passphrase, keyCipher, keyHash, keySize, pbkdf st
 }
 
 // OpenVolume opens volume so that it can be used by the client.
-func OpenVolume(volume, devicePath, passphrase string) error {
-	devPath := types.GetVolumeDevicePath(volume, true)
-	if isOpen, _ := IsDeviceOpen(devPath); isOpen {
-		logrus.Debugf("Device %s is already opened at %s", devicePath, devPath)
+// devicePath is the path of the volume on the host that will be opened for instance '/dev/longhorn/volume1'
+func OpenVolume(volume, dataEngine, devicePath, passphrase string) error {
+	encryptedDevName := types.GetEncryptVolumeName(volume, dataEngine)
```

Comment:
Why `types.GetEncryptVolumeName(volume, dataEngine)`?
The input of `IsDeviceOpen` is device path rather than a device name.

Reply:
`IsDeviceOpen` will finally use the `cryptsetup status [device]`, and [device] can be `[volume-name]` or `/dev/mapper/[volume-name]`. They are the same.
- `types.GetVolumeDevicePath(volume, dataEngine, true)` gets `/dev/mapper/[volume]-encrypted`
- `types.GetEncryptVolumeName(volume, dataEngine)` gets `[volume]-encrypted`

Reply:
Reverted it to use the full path. Thanks.